cpp/core.cc

```diff
@@ -254,6 +254,26 @@ bool InputAvailable(int fd) {
   return select(FD_SETSIZE, &fds, NULL, NULL, &timeout) > 0;
 }
 
+List<int>* WaitForInputs(List<int>* fds_in) {
+  auto* ret = NewList<int>();
+  fd_set fds;
+  FD_ZERO(&fds);
+  int n = len(fds_in);
+  for (int i = 0; i < n; ++i) {
+    FD_SET(fds_in->at(n), &fds);
+  }
+  int out = select(FD_SETSIZE, &fds, NULL, NULL, NULL);
+  for (int i = 0; i < n; ++i) {
+    if (InputAvailable(fds_in->at(n))) {
+      ret->append(fds_in->at(n));
```

Comment:
Hm I wonder if this is causing the IndexError

I don't think you should call `InputAvailable()` here ... the `select()` is supposed to return the list of descriptors

I think you make a new list from what select "returns", unconditionally